### PR TITLE
Kosta/znk chat correct paths

### PR DESCRIPTION
**Pull request by kostamas**

### CODE REVIEW

File: src/components/znkChat/templates/chatter.template.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div class="chatter-wrapper"
+     ng-class="{'offline': chatterObj.presence === d.userStatus.OFFLINE,
+     'online': chatterObj.presence === d.userStatus.ONLINE,
+     'idle': chatterObj.presence === vm.userStatus.IDLE}">
+    <div class="online-indicator"></div>
+    <div class="chatter-name">{{chatterObj.name}}</div>
+    <div class="message-not-seen"
+         ng-class="{'ten-or-more-unseen-messages': chatterObj.messagesNotSeen > 9}"
```
Comment (assafshp) [inline src/components/znkChat/templates/chatter.template.html]: 9 should be const

File: src/components/znkChat/svg/znk-chat-chat-icon.svg [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
+</g>
+<g>
```
Comment (assafshp) [inline src/components/znkChat/svg/znk-chat-chat-icon.svg]: why do you need all <g>

File: demoShared/storagesConfig.js [hunk 3]
```diff
@@ -85,6 +86,7 @@
             };
         }
 
-        InfraConfigSrvProvider.setStorages(storageGetter(options.dataDbPath), storageGetter(options.dataDbPath + options.studentPath));
+        // todo - added teacher storage
```
Comment (assafshp) [inline demoShared/storagesConfig.js]: what is this todo?

File: src/components/teachers/teachersSrv.service.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+     * */
+
+    angular.module('znk.infra.teachers').service('teachersSrv',
+        function (InfraConfigSrv) {
+            'ngInject';
+
+            var self = this;
+            var INVITATION_PATH = "users/$$uid/invitations/approved";
+
+            self.getAllTeachers = function () {
```
Comment (assafshp) [inline src/components/teachers/teachersSrv.service.js]: what will happen if you have no teachers?

File: src/components/teachers/teachersSrv.service.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+                    return studentStorage.get(INVITATION_PATH).then(function (teachers) {
+                        return _createNewTeachersObj(teachers);
+                    });
+                });
+            };
+
+            self.getTeacher = function (teacherUid) {
+                self.getAllTeachers().then(function (allTeachers) {
+                    if (angular.isDefined(allTeachers[teacherUid])) {
+                        return allTeachers;
```
Comment (assafshp) [inline src/components/teachers/teachersSrv.service.js]: are you sure it is not a bug ? I expect
return allTeachers[teacherUid]

Comment (assafshp) [inline src/components/teachers/teachersSrv.service.js]: what kind of object does this function return?

File: src/components/znkChat/directives/chat-message.directive.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+(function (angular) {
+    'use strict';
+
+    angular.module('znk.infra.znkChat').directive('chatMessage', [
+        function () {
+            'ngInject';
+            return {
+                template: '<div class="message-wrapper">' +
+                '<div class="message-date" ng-if="d.date">{{d.date}}</div>' +
```
Comment (assafshp) [inline src/components/znkChat/directives/chat-message.directive.js]: why do you need ng-if here?

File: src/components/znkChat/templates/chatBoard.template.html [hunk 7]
```diff
@@ -0,0 +1,28 @@
+<div class="chat-board-wrapper">
+    <div class="chat-board-header">
+        <chatter ng-if="chatterObj.uid" chatter-obj="chatterObj"></chatter>
+        <svg-icon name="znk-chat-close-icon" ng-click="d.closeChat()()"></svg-icon>
```
Comment (assafshp) [inline src/components/znkChat/templates/chatBoard.template.html]: ()()

File: src/components/znkChat/services/znkChatSrv.service.js [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+                };
+                newChatObj.uids[secondCUser.uid] = {
+                    isTeacher: secondCUser.isTeacher
+                };
+                newChatObj.usersLastSeenMessage = {};
+                newChatObj.usersLastSeenMessage[firstUser.uid] = {
+                    time: 0
+                };
+                newChatObj.usersLastSeenMessage[secondCUser.uid] = {
+                    time: 0,
```
Comment (assafshp) [inline src/components/znkChat/services/znkChatSrv.service.js]: remove ,

File: src/components/teachers/teachersSrv.service.js (hunk 4 above)
Comment (kostamas) [inline src/components/teachers/teachersSrv.service.js]: i fixed that- i return null if there is no teachers

File: src/components/znkChat/directives/chat-message.directive.js (hunk 6 above)
Comment (kostamas) [inline src/components/znkChat/directives/chat-message.directive.js]: there is a logic that show the message date only once (per  a day)

**Pull request merged**